app/models/job_proxy_dispatcher.rb

```diff
@@ -197,11 +263,29 @@ def busy_proxies
     end
   end
 
-  def active_scans
-    @active_scans ||= begin
-      actives = Hash.new(0)
-      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active").where.not(:state => "finished").count
-      actives
+  def active_scans_by_zone(target_class, count = true)
+    class_name = target_class.base_class.name
+    actives = Hash.new(0)
+    jobs = Job.where(:zone => MiqServer.my_zone, :dispatch_status => "active", :target_class => class_name)
```

Comment:
> why the change from @zone to MiqServer.my_zone?

If I'm reading the code right @zone always == MiqServer.my_zone after "initialization" in pending_jobs and when I added tests to this method it failed. I don't know @jrafanie  maybe initialization is better done in the test  and I should leave this untouched?